### ISSUE TITLE
Model updates aren't triggering useEffect(), field is types.array

### ISSUE THREAD

**Issue by pidogc** [mobxjs/mobx-state-tree#1854]
**_Question_**

* [x] I've checked documentation and searched for existing issues
* [x] I tried [GitHub Discussions](https://github.com/mobxjs/mobx-state-tree/discussions)

this is my code ,Not sure what wrong

checkoutStore.timesForDeliveryOrPickup this is array

![image](https://user-images.githubusercontent.com/42857270/149440745-bb422e0e-5ff8-4db0-a71b-6eac0f0dd0f4.png)

**Comment by EmilTholin** [comment]
Hi @pidogc!

It's hard to say exactly what's wrong without seeing a small runnable example that illustrates your problem, but it might be that `checkoutStore.timesForDeliveryOrPickup` it not actually dereferenced in your component.

You can write `checkoutStore.timesForDeliveryOrPickup.slice()` to dereference the array so that the observer component rendering `PaymentMethod` will re-render when `checkoutStore.timesForDeliveryOrPickup` changes.

**Comment by pidogc** [comment]
@EmilTholin thx! After I use the toJS method of mobx, the update can be triggered normally. Is this conversion the correct way to use it?

**Comment by EmilTholin** [comment]
Nice! Yeah, `toJS` should give the same end result in this case I believe, but it might be a bit more costly than just `checkoutStore.timesForDeliveryOrPickup.slice()`.

**Comment by pidogc** [comment]
@EmilTholin Thank you for your help!

**Comment by longnguyen2508** [comment]
This one should be a documentary

**Comment by coolsoftwaretyler** [comment]
Hi @longnguyen2508 - can you say more about what you mean?

**Comment by zcmgyu** [comment]
Hi @coolsoftwaretyler 
I’m using mobx-state-tree without mobx knowledge. Consequently, I understand that any updates to the data in the store will be mirrored through the object retrieved from views within the Component that is encapsulated by the observer API. However, I encountered the same problem, indicating that I needed to transform it into JavaScript using the toJS API or slice API. Despite reviewing the documentation, I found no guidance on the proper application of these APIs within React.
